### ISSUE TITLE
Indenting function/struct within a module causes `invalid-function` error

### ISSUE THREAD

**Issue by Linerre** [amnn/move-mode#11]
Dear author,

After upgrading to v1.0.0 recently, I ran into this problem when I tried to indent functions within a module. 
As you can see from the screenshot below, if I try to press `TAB` to indent the line the point is currently positioned, I got the error (also captured in the screenshot):
```
Debugger entered--Lisp error: (invalid-function (find-annotation (open-parens) (cond ((not open-parens) nil) ((char-equal 40 (char-after (car open-parens))) (find-annotation (cdr open-parens))) ((char-equal 91 (char-after (car open-parens))) (goto-char (car open-parens)) (backward-char) (looking-at "#\\[")) (t nil))))
  (find-annotation (open-parens) (cond ((not open-parens) nil) ((char-equal 40 (char-after (car open-parens))) (find-annotation (cdr open-parens))) ((char-equal 91 (char-after (car open-parens))) (goto-char (car open-parens)) (backward-char) (looking-at "#\\[")) (t nil)))()
  move--ppss-in-annotation()
  move--prev-assignment(74)
  move--indent-column()
  move-mode-indent-line()
  indent--funcall-widened(move-mode-indent-line)
  indent-for-tab-command(nil)
  funcall-interactively(indent-for-tab-command nil)
  call-interactively(indent-for-tab-command nil nil)
  command-execute(indent-for-tab-command)
```

![image](https://github.com/user-attachments/assets/123b9a7e-9cc3-49dd-bcf3-174b38b58007)

Output of `M-x emacs-version`:
```
GNU Emacs 29.4 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.43, cairo version 1.18.2)
```

I installed `move-mode` using `straight.el` and I'm on the latest commit of the main branch: `f974cc69f279c45026f7386e0194be74779334a`

My `move-mode` config is pretty simple
```
(autoload 'move-mode "move-mode" nil t)
(add-to-list 'auto-mode-alist '("\\.move\\'" . move-mode))
(with-eval-after-load "move-mode"
  (define-key move-mode-map (kbd"C-c C-l") #'lsp))
```

Indenting a field of a struct has no problems.  Could you take a look at the error? Many thanks

**Comment by Linerre** [comment]
perhaps the required library is missing in the `move-mode.el`?
```
(require 'cl-lib)
```

**Comment by amnn** [comment]
Hi @Linerre, thanks for the report and sorry you hit this issue! I think you're right that we're missing a require for `cl-lib` so `cl-labels` isn't being recognised as a macro. If you're happy to submit a PR to add that, I will merge it, otherwise I'll work on reproducing the issue on a minimal config here and fixing.

**Comment by Linerre** [comment]
> If you're happy to submit a PR to add that, I will merge it, otherwise I'll work on reproducing the issue on a minimal config here and fixing.

Hi @amnn, thanks for your reply. I'd like to but I'm not 100% sure about the cause of this issue. So I have tried the following with zero-configured Emacs and just the `move-mode.el`:
```
emacs -q -L . -l move-mode.el
```
where `.` means I'm inside the `move-mode` source code directory on my machine. When trying to indent the `swords_created` function, I got `Symbol's function definition is void: cl-lables` in the echo area:

![image](https://github.com/user-attachments/assets/732013c7-5633-4565-92a2-7b93c33b4c28)

The code I used for testing is from [Sui's official doc](https://docs.sui.io/guides/developer/first-app/write-package#defining-the-package) and I'm pasting it here just for convenience

```move
module entry_functions::example {
    use sui::object::{Self, UID};
    use sui::transfer;
    use sui::tx_context::{Self, TxContext};

    struct Foo has key {
        id: UID,
        bar: u64,
    }

    /// Entry functions can accept a reference to the `TxContext`
    /// (mutable or immutable) as their last parameter.
    entry fun share(bar: u64, ctx: &mut TxContext) {
        transfer::share_object(Foo {
            id: object::new(ctx),
            bar,
        })
    }

    /// Parameters passed to entry functions called in a programmable
    /// transaction block (like `foo`, below) must be inputs to the
    /// transaction block, and not results of previous transactions.
    entry fun update(foo: &mut Foo, ctx: &TxContext) {
        foo.bar = tx_context::epoch(ctx);
    }

    /// Entry functions can return types that have `drop`.
    entry fun bar(foo: &Foo): u64 {
        foo.bar
    }

    /// This function cannot be `entry` because it returns a value
    /// that does not have `drop`.
    public fun foo(ctx: &mut TxContext): Foo {
        Foo { id: object::new(ctx), bar: 0 }
    }
}
```

If I add the following line, both `move-mode.el` and `move-mode.elc` (re-compiled using `straight-rebuild-package`) work for the indentation:
```
(require 'cl-lib)
```

I have created a PR anyway, but feel free to test/reproduce/check whichever part you have doubt about. Thank you for this awesome mode package :)

**Comment by amnn** [comment]
Thanks @Linerre! I did try to repro on my machine, using Emacs 29.4 and Emacs 30.6 (which is what I have installed):

```
GNU Emacs 30.0.60 (build 1, aarch64-apple-darwin23.5.0, NS appkit-2487.60 Version 14.5 (Build 23F79)) of 2024-07-16
```

Neither exhibited the issue on a bare config, so I guess something else is loading `cl-lib` during startup on my build. In any case, I also tested your fix and there were no issues with that either, so happy to merge, thanks for the super quick turnaround!
